app/va/va_profile/va_profile_client.py

```diff
@@ -165,35 +170,38 @@ def get_email(self, va_profile_id: RecipientIdentifier) -> str:
         self.statsd_client.incr('clients.va-profile.get-email.failure')
         self._raise_no_contact_info_exception(self.EMAIL_BIO_TYPE, va_profile_id, contact_info.get(self.TX_AUDIT_ID))
 
-    def has_valid_mobile_telephone_classification(self, telephone: Telephone) -> bool:
+    def has_valid_mobile_telephone_classification(self, telephone: Telephone, contact_info: ContactInformation) -> bool:
         """
         Args:
           telephone (Telephone): telephone entry from ContactInformation object retrieved from Vet360 API endpoint
 
         Returns:
           bool - if AWS-classified telephone is a valid sms recipient (if nonexistent, return True)
+
+        Raises:
+          InvalidPhoneNumberException - if AWS-classified telephone is not a valid sms recipient
         """
         classification = telephone.get('classification', {})
         classification_code = classification.get('classificationCode', None)
-        if classification_code is not None:
-            if classification_code not in VALID_PHONE_TYPES_FOR_SMS_DELIVERY:
-                self.logger.debug(
-                    'V3 Profile -- Phone classification code of %s is not a valid SMS recipient (VA Profile ID: %s)',
-                    classification_code,
-                    telephone['vaProfileId'],
-                )
-                return False
+        if classification_code is None:
+            # fall back, if no phone number classification is present
+            self.logger.debug(
+                'V3 Profile -- No telephone classification present, assuming the number is a valid SMS recipient (VA Profile ID: %s)',
+                telephone['vaProfileId'],
+            )
+            return True
 
+        if classification_code not in VALID_PHONE_TYPES_FOR_SMS_DELIVERY:
             self.logger.debug(
-                'V3 Profile -- Phone classification code of %s is a valid SMS recipient (VA Profile ID: %s)',
+                'V3 Profile -- Phone classification code of %s is not a valid SMS recipient (VA Profile ID: %s)',
                 classification_code,
                 telephone['vaProfileId'],
             )
-            return True
+            self._raise_invalid_phone_number_exception(contact_info)
 
-        # fall back, if no phone number classification is present
         self.logger.debug(
-            'V3 Profile -- No telephone classification present, assuming the number is a valid SMS recipient (VA Profile ID: %s)',
+            'V3 Profile -- Phone classification code of %s is a valid SMS recipient (VA Profile ID: %s)',
+            classification_code,
             telephone['vaProfileId'],
         )
         return True
```

Comment:
This is a possible rewrite of this function to make it a little more clear what's going on in the future. You would have to move the exception being raised to the calling function in this case, so this one always returns a boolean value.

```suggestion
        is_valid_number = False
        classification = telephone.get('classification', {})
        classification_code = classification.get('classificationCode', None)
        if classification_code is None:
            # fall back, if no phone number classification is present
            self.logger.debug(
                'V3 Profile -- No telephone classification present, assuming the number is a valid SMS recipient (VA Profile ID: %s)',
                telephone['vaProfileId'],
            )
            is_valid_number = True
        elif classification_code in VALID_PHONE_TYPES_FOR_SMS_DELIVERY:
            self.logger.debug(
                'V3 Profile -- Phone classification code of %s is a valid SMS recipient (VA Profile ID: %s)',
                classification_code,
                telephone['vaProfileId'],
            )
            is_valid_number = True
        else:
            self.logger.debug(
                'V3 Profile -- Phone classification code of %s is not a valid SMS recipient (VA Profile ID: %s)',
                classification_code,
                telephone['vaProfileId'],
            )

        return is_valid_number
```

Reply:
Hey Evan I'm planning on cleaning this code up once we are ready to remove the feature toggles.  Right now this code is structured this way in order to allow us to do a bunch of logging.  In the future, we'll probably only log on error state.